src/network/networkd-sysctl.c

```diff
@@ -5,17 +5,193 @@
+        return NULL;
+}
+
+static struct ring_buffer *rb_free(struct ring_buffer *rb) {
+        sym_ring_buffer__free(rb);
+        return NULL;
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct sysctl_monitor_bpf *, sysctl_monitor_bpf_free);
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct ring_buffer *, rb_free);
+
+static int sysct_write_event(void *ctx, void *data, size_t data_sz) {
+        struct sysctl_write_event *we = ASSERT_PTR(data);
+        Hashmap **sysctl_shadow = ASSERT_PTR(ctx);
+        _cleanup_free_ char *path = NULL;
+        char *value;
+
+        if (we->version != 1)
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                "Invalid sysctl write event version: %d", we->version);
+
+        path = path_join("/proc/sys", we->name);
+        if (!path)
+                return log_oom();
+
+        /* If we never managed this handle, ignore it. */
+        value = hashmap_get(*sysctl_shadow, path);
+        if (!value)
+                return 0;
+
+        if (!streq(value, we->newvalue))
+                log_struct(LOG_WARNING,
+                           LOG_MESSAGE("Foreign process '%s' changed sysctl '%s' from '%s' to '%s', conflicting with our setting to '%s'",
+                                        we->comm, we->name, we->current, we->newvalue, value),
+                           "MESSAGE_ID=" SD_MESSAGE_SYSCTL_CHANGED_STR);
+
+        return 0;
+}
+
+static int on_ringbuf_io(sd_event_source *s, int fd, uint32_t revents, void *userdata) {
+        struct ring_buffer *rb = ASSERT_PTR(userdata);
+        int r;
+
+        r = sym_ring_buffer__poll(rb, /* timeout_msec= */ 0);
+        if (r < 0 && errno != EINTR)
+                log_error_errno(errno, "Error polling ring buffer: %m");
+
+        return 0;
+}
+
+int sysctl_add_monitor(Manager *manager) {
+        _cleanup_(sysctl_monitor_bpf_freep) struct sysctl_monitor_bpf *obj = NULL;
+        _cleanup_(bpf_link_freep) struct bpf_link *sysctl_link = NULL;
+        _cleanup_(rb_freep) struct ring_buffer *sysctl_buffer = NULL;
+        _cleanup_close_ int cgroup_fd = -EBADF, rootcg = -EBADF;
+        _cleanup_free_ char *cgroup = NULL;
+        int idx = 0;
+        int r;
+
+        assert(manager);
+
+        r = dlopen_bpf();
+        if (r < 0) {
+                log_info_errno(r, "sysctl monitor disabled, as BPF support is not available.");
+                return 0;
+        }
+
+        r = cg_pid_get_path(SYSTEMD_CGROUP_CONTROLLER, 0, &cgroup);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to get cgroup path: %m, ignoring.");
+
+        rootcg = cg_path_open(SYSTEMD_CGROUP_CONTROLLER, "/");
+        if (rootcg < 0)
+                return log_warning_errno(rootcg, "Failed to open cgroup: %m, ignoring.");
+
+        obj = sysctl_monitor_bpf__open_and_load();
+        if (!obj) {
+                log_info_errno(errno, "Failed to load sysctl monitor BPF program: %m, ignoring.");
```

Comment:
given that we don't consider this an error we should say "fail" here. Maybe tone it down to "Unable to"